lib/kafka_ex/protocol/metadata.ex

```diff
@@ -113,13 +143,55 @@ defmodule KafkaEx.Protocol.Metadata do
     parse_brokers(brokers_size - 1, rest, [%Broker{node_id: node_id, host: host, port: port} | brokers])
   end
 
+  defp parse_brokers_v1(0, rest, brokers), do: {brokers, rest}
```

Comment:
Attaching version numbers to the functions will become cumbersome, I think. Many of the APIs have several versions, but those version might not have any differences, which might lend itself better to having one a function parameter that is the API version.

For example, Metadata Response has no differences in the `brokers` field from versions 1-6.

Reply:
Indeed. I think it's going to become cumbersome either way, to be honest :).

I've switched to using a function parameter instead. Tell me what you think !